test-case/test-speaker.sh

```diff
@@ -33,26 +35,23 @@ logger_disabled || func_lib_start_log_collect
 func_pipeline_export "$tplg" "type:playback"
 tcnt=${OPT_VAL['l']}
 setup_kernel_check_point
-for idx in $(seq 0 $(expr $PIPELINE_COUNT - 1))
+for idx in $(seq 0 $((PIPELINE_COUNT - 1)))
 do
-    channel=$(func_pipeline_parse_value $idx channel)
-    rate=$(func_pipeline_parse_value $idx rate)
-    fmt=$(func_pipeline_parse_value $idx fmt)
-    dev=$(func_pipeline_parse_value $idx dev)
-    snd=$(func_pipeline_parse_value $idx snd)
+    channel=$(func_pipeline_parse_value "$idx" channel)
+    rate=$(func_pipeline_parse_value "$idx" rate)
+    fmt=$(func_pipeline_parse_value "$idx" fmt)
+    dev=$(func_pipeline_parse_value "$idx" dev)
+    snd=$(func_pipeline_parse_value "$idx" snd)
 
     dlogc "speaker-test -D $dev -r $rate -c $channel -f $fmt -l $tcnt -t wav -P 8"
-    speaker-test -D $dev -r $rate -c $channel -f $fmt -l $tcnt -t wav -P 8 2>&1 |tee $LOG_ROOT/result_$idx.txt
-    resultRet=$?
+    speaker-test -D "$dev" -r "$rate" -c "$channel" -f "$fmt" -l "$tcnt" -t wav -P 8 2>&1 |tee "$LOG_ROOT"/result_"$idx".txt
+    resultRet=${PIPESTATUS[0]} 
 
-    if [[ $resultRet -eq 0 ]]; then
-        grep -nr -E "error|failed" $LOG_ROOT/result_$idx.txt
-        if [[ $? -eq 0 ]]; then
-            func_lib_lsof_error_dump $snd
+    if [[ $resultRet -ne 0  ]] ||
+        grep -nr -E "error|failed" "$LOG_ROOT"/result_"$idx".txt; then
```

Comment:
Swap the conditions to  make sure the error messages are always highlighted

```suggestion
    if  grep -nr -E "error|failed" "$LOG_ROOT"/result_"$idx".txt ||
       [[ $resultRet -ne 0  ]] ;  then
```